src/res_scanner.ml

```diff
@@ -133,23 +133,27 @@ let make ~filename src =
     mode = [];
   }
 
+let isWhitespace ch =
+  match ch with
+  | ' ' | '\t' | '\n' | '\r' -> true
+  | _ -> false
+
 let rec skipWhitespace scanner =
-  match scanner.ch with
-  | ' ' | '\t'| '\n' | '\r' ->
+  if isWhitespace scanner.ch then (
     next scanner;
     skipWhitespace scanner
-  | _ -> ()
+  )
 
 let scanIdentifier scanner =
   let startOff = scanner.offset in
-  while (
-    CharacterCodes.isLetter scanner.ch ||
-    CharacterCodes.isDigit scanner.ch ||
-    '_' == scanner.ch ||
-    '\'' == scanner.ch
-  ) do
-    next scanner
-  done;
+  let rec skipGoodChars scanner =
```

Comment:
better assembly?

Reply:
Vs while loop

Reply:
No and I'm not optimizing at this level anymore, but you'll see